interop-monix/shared/src/test/scala/zio/interop/monix/MonixTaskSpec.scala

```diff
@@ -0,0 +1,162 @@
+package zio
+package interop.monix
+
+import duration._
+
+import test._
+import Assertion._
+import TestAspect._
+
+import _root_.monix.eval.{ Task => MTask }
+
+object MonixTaskSpec extends DefaultRunnableSpec {
+
+  override def spec =
+    suite("MonixTaskSpec")(
+      suite("ZIO.fromMonixTask")(
+        testM("return a ZIO that fails if Monix task failed") {
+          val error     = new Exception("monix task failed")
+          val monixTask = MTask.raiseError(error)
+          val io        = ZIO.fromMonixTask(monixTask)
+
+          assertM(io.either)(isLeft(equalTo(error)))
+        },
+        testM("return a ZIO that succeeds if Monix task succeeded") {
+          val result    = "monix task result"
+          val monixTask = MTask.now(result)
+          val io        = ZIO.fromMonixTask(monixTask)
+
+          assertM(io)(equalTo(result))
+        },
+        testM("converts a synchronous Monix task to a ZIO task") {
+          @volatile var testVar = 0
+          val monixTask         = MTask.eval {
+            val old = testVar
+            testVar = old + 1
+            old
+          }
+          val io                = for {
+            orig    <- ZIO.fromMonixTask(monixTask)
+            current <- ZIO.succeed(testVar)
+          } yield ((orig, current))
+          assertM(io)(equalTo((0, 1)))
+        },
+        testM("converts an asynchronous Monix task to a ZIO task") {
+          @volatile var testVar = 0
+          val monixTask         = MTask.async[Int] { cb =>
+            val old = testVar
+            testVar = old + 1
+            cb.onSuccess(old)
+          }
+          val io                = for {
+            orig    <- ZIO.fromMonixTask(monixTask)
+            current <- ZIO.succeed(testVar)
+          } yield ((orig, current))
+          assertM(io)(equalTo((0, 1)))
+        },
+        testM("the ZIO task fails if a Monix async task fails") {
+          val error     = new Exception("async monix failure")
+          val monixTask = MTask.async[Int] { cb =>
+            cb.onError(error)
+          }
+          val io        = ZIO.fromMonixTask(monixTask)
+          assertM(io.either)(isLeft(equalTo(error)))
+        },
+        testM("lazily executes a converted Monix task") {
+          @volatile var executed = false
+          val monixTask          = MTask.eval {
+            executed = true
+          }
+          val test               = ZIO.succeed {
+            executed = false
+          } *> {
+            ZIO.fromMonixTask(monixTask)
+            ZIO.succeed(executed)
```

Comment:
Should these two be chained? Also, `ZIO.succeed` => `UIO`.

Reply:
I'm trying to prove that the Monix task isn't executed just by creating the ZIO value wrapping it, we'd have to actually run the ZIO effect. I couldn't think of a better way.